spec/type/security_group_spec.rb

```diff
@@ -1,31 +1,65 @@
 require 'spec_helper'
+require 'awspec/error'
+
 Awspec::Stub.load 'security_group'
 
 describe security_group('sg-1a2b3cd4') do
   it { should exist }
-  its(:inbound) { should be_opened(80) }
+  it do
+    # only way found to force and test the exception
+    subject.inbound
+    expect { subject.opened?(nil) }.to raise_error(Awspec::MissingPortSpecification, /protocol\sis\s-1$/)
+  end
+  its(:inbound) { should_not be_opened(80) }
+  its(:inbound) { should_not be_opened(80).protocol(-1) }
+  its(:inbound) { should be_opened(80).protocol(-1).for('sg-3a4b5cd6') }
   its(:inbound) { should be_opened(80).protocol('tcp').for('123.45.68.89/32') }
   its(:inbound) { should be_opened(80).protocol('tcp').for('123.45.67.0/25') }
   its(:inbound) { should be_opened(80).protocol('tcp').for('123.45.67.1/32') }
   its(:inbound) { should_not be_opened(80).protocol('tcp').for('123.45.0.0/16') }
-  its(:inbound) { should be_opened(22) }
```

Comment:
I think that there are times when you just want to test that the port is open. So I'd like to see support for this notation as well.

Reply:
> but my guess is that is not acceptable to create a security group with such rule, without specifying protocol and CIDR.

Yes.

> Even if it would work, it would be a security flaw in my opinion: there is a big different about asserting that a port is opened for a specific IP or the whole internet (0.0.0.0).

Yes. This test does not specify the IP and protocol, so it will be to make sure that port 22 is open with some protocol on some IP.

Another way to say it is the negation of `should_not be_opened(22)`.